Pocket/content-monorepo
# feat(ProspectAPI): [MC-473] Send prospects to Firehose

## Goal
ML wants to access all generated prospects from Snowflake. Everyone is aligned on the longer term vision of replacing SQS with EventBridge, and with using Snowplow instead of Firehose to ingest into our Data Warehouse.

Currently, prospects generated in AWS get to Snowflake through Firehose. To unblock prospecting scheduler items we need to keep prospects flowing through Firehose.

This PR ingests prospects from GCP to Snowflake by sending them to the Firehose stream.

## I'd love feedback/perspectives on:
- How do we handle Lambda Node updates? Should we aim to have all code in content-monorepo run on the same Node version? My inclination would be 'yes'. Given that we have a single .nvmrc, we are already developing and running tests on Node 20, even though the Translation Lambda is configured to use Node 14.
- Should dependencies in package.json be pinned as "^3.504.0" or "3.504.0"? It's not consistent now. I think the former?

## References

JIRA ticket:

- https://mozilla-hub.atlassian.net/browse/MC-473

Issue:

- Link to GitHub issue

Documentation:

- [Somewhat relevant Slack thread](https://mozilla.slack.com/archives/C05MEMLP943/p1706738387476009?thread_ts=1706728938.133769&cid=C05MEMLP943):

Somewhat related to this work, Gaurang said:
> The curation tool currently only sends dismissed prospects as Snowplow events.
> [...]
> If the tool sends prospects as snowplow events, we could eventually de-commission the legacy method of sending prospects as candidate sets [through Firehose]

## QA

Steps:
- Open [ProspectAPI-Dev-Sqs-Bridge-Queue](https://us-east-1.console.aws.amazon.com/sqs/v3/home?region=us-east-1#/queues/https%3A%2F%2Fsqs.us-east-1.amazonaws.com%2F410318598490%2FProspectAPI-Dev-Sqs-Bridge-Queue/send-receive) in the dev account
- Send a message with body set to [prospect_set.json](https://github.com/Pocket/content-monorepo/files/14068363/prospect_set.json)

Expectations:
- [x] [Bridge Lambda](https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/ProspectAPI-Dev-Sqs-Bridge-Function?tab=code) and [Translation Lambda](https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/ProspectAPI-Dev-Sqs-Translation-Function?tab=code) are 'Node.js 20.x'
- [x] [ProspectAPI-Dev-Sqs-Bridge-Function](https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/ProspectAPI-Dev-Sqs-Bridge-Function?tab=monitoring) shows a successful invocation
- [x] Search in the [ProspectAPI-Dev-Sqs-Translation-Function logs for "38 prospects inserted into dynamo"](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:log-groups/log-group/$252Faws$252Flambda$252FProspectAPI-Dev-Sqs-Translation-Function/log-events$3Fstart$3D-1800000$26filterPattern$3D$252238+prospects+inserted+into+dynamo$2522), and check that the timestamp is within a few minutes after you sent the SQS message.
